go/common/viewingkey/viewing_key.go

```diff
@@ -33,41 +32,45 @@ type RPCSignedViewingKey struct {
 }
 
 // GenerateViewingKeyForWallet takes an account wallet, generates a viewing key and signs the key with the acc's private key
-// uses the same method of signature handling as Metamask/geth
-// TODO @Ziga - update this method to use the new EIP-712 signature format / personal sign after the removal of the legacy format
 func GenerateViewingKeyForWallet(wal wallet.Wallet) (*ViewingKey, error) {
-	// generate an ECDSA key pair to encrypt sensitive communications with the obscuro enclave
-	vk, err := crypto.GenerateKey()
+	chainID := int64(443)
```

Comment:
I think wallet has a chainID field on it, might be better to use that rather than hardcoding if poss. In case we end up with different chainIDs on diff Ten networks eventually.

Reply:
nice, fixed 👍